config/config_test.go

```diff
@@ -0,0 +1,205 @@
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	validFile = `
+{
+  "Servers": {
+    "rancherDefault": {
+      "accessKey": "the-access-key",
+      "secretKey": "the-secret-key",
+      "tokenKey": "the-token-key",
+      "url": "https://example.com",
+      "project": "cluster-id:project-id",
+      "cacert": "",
+      "kubeCredentials": null,
+      "kubeConfigs": null
+    }
+  },
+  "CurrentServer": "rancherDefault"
+}`
+	invalidFile = `invalid config file`
+)
+
+func Test_CheckPermission(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name             string
+		mode             os.FileMode
+		expectedWarnings int
+	}{
+		{
+			name:             "neither group-readable nor world-readable",
+			mode:             os.FileMode(0600),
+			expectedWarnings: 0,
+		},
+		{
+			name:             "group-readable and world-readable",
+			mode:             os.FileMode(0644),
+			expectedWarnings: 2,
+		},
+		{
+			name:             "group-readable",
+			mode:             os.FileMode(0640),
+			expectedWarnings: 1,
+		},
+		{
+			name:             "world-readable",
+			mode:             os.FileMode(0604),
+			expectedWarnings: 1,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			assert := assert.New(t)
+
+			dir, err := os.MkdirTemp("", "rancher-cli-test-*")
+			assert.NoError(err)
+			defer os.RemoveAll(dir)
+
+			path := filepath.Join(dir, "cli2.json")
+			os.WriteFile(path, []byte(validFile), tt.mode)
```

Comment:
You're not really checking the error:
```suggestion
			err = os.WriteFile(path, []byte(validFile), tt.mode)
			assert.NoError(err)
```

Reply:
Good find!